cli/spinner.ts

```diff
@@ -0,0 +1,139 @@
+// Copyright 2018-2023 the Deno authors. All rights reserved. MIT license.
+
+// TODO(kt3k): Write test when pty is supported in Deno
+// See: https://github.com/denoland/deno/issues/3994
+
+const encoder = new TextEncoder();
+
+const LINE_CLEAR = encoder.encode("\r\u001b[K"); // From cli/prompt_secret.ts
+const COLOR_RESET = "\u001b[0m";
+const DEFAULT_INTERVAL = 75;
+const DEFAULT_SPINNER = ["⠋", "⠙", "⠹", "⠸", "⠼", "⠴", "⠦", "⠧", "⠇", "⠏"];
+
+// This is a hack to allow us to use the same type for both the color name and an ANSI escape code.
+// ref: https://github.com/microsoft/TypeScript/issues/29729#issuecomment-460346421
+// deno-lint-ignore ban-types
+type Ansi = string & {};
+type Color =
+  | "black"
+  | "red"
+  | "green"
+  | "yellow"
+  | "blue"
+  | "magenta"
+  | "cyan"
+  | "white"
+  | "gray"
+  | Ansi;
+
+const COLORS: Record<Color, string> = {
+  black: "\u001b[30m",
+  red: "\u001b[31m",
+  green: "\u001b[32m",
+  yellow: "\u001b[33m",
+  blue: "\u001b[34m",
+  magenta: "\u001b[35m",
+  cyan: "\u001b[36m",
+  white: "\u001b[37m",
+  gray: "\u001b[90m",
+};
+
+/** Options for {@linkcode Spinner}. */
+export interface SpinnerOptions {
+  /**
+   * The sequence of characters to be iterated through for animation.
+   *
+   * @default {["⠋", "⠙", "⠹", "⠸", "⠼", "⠴", "⠦", "⠧", "⠇", "⠏"]}
+   */
+  spinner?: string[];
+  /** The message to display next to the spinner. */
+  message?: string;
+  /** The time between each frame of the spinner.
```

Comment:
It would be helpful to include the unit of time here (milliseconds) — otherwise, the user must review the implementation to understand. For example:

```diff
- The time between each frame of the spinner.
+ The time (milliseconds) between each frame of the spinner.
```